pkg/controller/registry/resolver/steps.go

```diff
@@ -118,6 +118,15 @@ func NewStepResourceFromBundle(bundle *api.Bundle, namespace, replaces, catalogS
 		return nil, err
 	}
 
+	// Check unpacked bundled for for missing APIVersion or Kind
+	if csv.APIVersion == "" {
+		return nil, fmt.Errorf("bundle CSV %s missing APIVersion", csv.Name)
+	}
+
+	if csv.Kind == "" {
```

Comment:
Following up on this comment that I missed. Question before I implement your suggested change - does it maybe make sense from a UX perspective to return an error here anyway so that there is consistent behavior when a field is missing from a CSV?

Reply:
I think it's reasonable for the sake of consistency to return an error -- it's doesn't make total sense, since we are handling a concrete CSV type in the code (which we inferred somehow from the underlying JSON representation of the CSV in the bundle) to then turn around and say "this CSV Kind is undefined", but I guess it's simply a matter of preference: silently accepting the manifest (which I argue a smart package manager should be doing) versus erroring. But I'm ok with staying consistent. 

/lgtm